include/deal.II/base/table_indices.h

```diff
@@ -81,294 +109,79 @@ class TableIndicesBase
 
-  /**
-   * Constructor. Set indices to the given values.
-   *
-   * The default values for the second and subsequent arguments are necessary
-   * for some neat template tricks in SymmetricTensor where we only want to
-   * set the first index and construct the subsequent ones later on, i.e. for
-   * the moment we don't care about the later indices.
-   */
-  TableIndices (const unsigned int index1,
-                const unsigned int index2 = numbers::invalid_unsigned_int,
-                const unsigned int index3 = numbers::invalid_unsigned_int,
-                const unsigned int index4 = numbers::invalid_unsigned_int,
-                const unsigned int index5 = numbers::invalid_unsigned_int,
-                const unsigned int index6 = numbers::invalid_unsigned_int);
-};
+  for (unsigned int i=0; i<N; ++i)
+    indices[i] = 0;
+}
 
 
-/**
- * Array of indices of fixed size used for the TableBase class.
- *
- * This is the specialization for a seven-dimensional table. This class only
- * differs in the non-default constructors from the other specializations.
- * Actual storage of and access to data is done by the TableIndicesBase base
- * class of a specializations.
- *
- * @ingroup data
- * @author Wolfgang Bangerth, 2002, Ralf Hartmann 2004
- */
-template <>
-class TableIndices<7> : public TableIndicesBase<7>
+template <int N>
+TableIndices<N>::TableIndices(const unsigned int index0,
+                              const unsigned int index1,
+                              const unsigned int index2,
+                              const unsigned int index3,
+                              const unsigned int index4,
+                              const unsigned int index5,
+                              const unsigned int index6,
+                              const unsigned int index7,
+                              const unsigned int index8)
 {
-public:
-  /**
-   * Default constructor. Set all indices to zero.
-   */
-  TableIndices ();
-
-  /**
-   * Constructor. Set indices to the given values.
-   *
-   * The default values for the second and subsequent arguments are necessary
-   * for some neat template tricks in SymmetricTensor where we only want to
-   * set the first index and construct the subsequent ones later on, i.e. for
-   * the moment we don't care about the later indices.
-   */
-  TableIndices (const unsigned int index1,
-                const unsigned int index2 = numbers::invalid_unsigned_int,
-                const unsigned int index3 = numbers::invalid_unsigned_int,
-                const unsigned int index4 = numbers::invalid_unsigned_int,
-                const unsigned int index5 = numbers::invalid_unsigned_int,
-                const unsigned int index6 = numbers::invalid_unsigned_int,
-                const unsigned int index7 = numbers::invalid_unsigned_int);
-};
+  Assert (N != 0, ExcMessage("Cannot create a TableIndices object of size 0"));
```

Comment:
My comment about checking that unused arguments remain at the default value is gone, and so is your answer. The email that was sent out also didn't include the example that you wanted to show, but in the end I'm still in favor of such assertions -- passing wrong values should simply *never* be allowed. It is almost always a sign of a bug lurking somewhere.